src/main/webapp/app/exercises/programming/shared/code-editor/ace/code-editor-ace.component.ts

```diff
@@ -108,10 +117,7 @@ export class CodeEditorAceComponent implements AfterViewInit, OnChanges, OnDestr
     annotationsArray: Array<Annotation> = [];
     annotationChange: Subscription;
     fileSession: FileSession = {};
-    // Inline feedback variables
-    fileFeedbacks: Feedback[];
-    fileFeedbackPerLine: { [line: number]: Feedback } = {};
-    linesWithNewFeedback: number[] = []; // lines with new feedback, which is not yet saved, but must already be displayed to be editable
+    newFeedbackLines: number[] = []; // new feedback, which is not yet saved, but must already be displayed to be editable
```

Comment:
Note for code reviewers: The file feedback is now mostly dynamically computed, which doesn't impact performance and is much more robust.